src/main/java/com/conveyal/datatools/manager/models/transform/AppendToFileTransformation.java

```diff
@@ -0,0 +1,81 @@
+
+package com.conveyal.datatools.manager.models.transform;
+
+import com.conveyal.datatools.common.status.MonitorableJob;
+import com.conveyal.datatools.manager.models.TableTransformResult;
+import com.conveyal.datatools.manager.models.TransformType;
+
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.FileSystem;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+
+public class AppendToFileTransformation extends ZipTransformation {
+
+    public static AppendToFileTransformation create(String csvData, String table) {
+        AppendToFileTransformation transformation = new AppendToFileTransformation();
+        transformation.csvData = csvData;
+        transformation.table = table;
+        return transformation;
+    }
+
+    @Override
+    public void validateParameters(MonitorableJob.Status status) {
+        if (csvData == null) {
+            status.fail("CSV data must not be null");
+        }
+    }
+
+    @Override
+    public void transform(FeedTransformZipTarget zipTarget, MonitorableJob.Status status) {
+        String tableName = table + ".txt";
+        Path targetZipPath = Paths.get(zipTarget.gtfsFile.getAbsolutePath());
+
+        try (
+            FileSystem targetZipFs = FileSystems.newFileSystem(targetZipPath, (ClassLoader) null);
+            InputStream newLineStream = new ByteArrayInputStream("\n".getBytes(StandardCharsets.UTF_8));
```

Comment:
Instead of creating an input stream for a single character, could the new line be prefixed to `csvData` on line 25?

Reply:
Unfortunately that method is only fired during the unit tests!